### PR TITLE
circuit.draw display when ipython

### PR DESCRIPTION
**Pull request by 1ucian0**
Thanks to @frankharkins, I learned about `IPython.display.display`. This allows circuit to be drew circuits in a loop.

Take the following example:
```python
from qiskit import QuantumCircuit
qc1 = QuantumCircuit(2)
qc1.h(0)
qc1.cx(0,1)

qc2 = QuantumCircuit(2)
qc2.h(1)
qc2.cx(1,0)

circuits = [qc1, qc2]
``` 

These are the outputs:
Screen Shot 2020-10-30 at 5 08 52 PM
Screen Shot 2020-10-30 at 5 09 00 PM
Screen Shot 2020-10-30 at 5 09 07 PM
Screen Shot 2020-10-30 at 5 09 16 PM

This makes #5030 unnecessary, so removing.

### CODE REVIEW

**Comment by frankharkins** [comment]
I think this would cause confusing behaviour. For example, in Jupyter Notebooks we wouldn't be able to do `qc.draw().savefig()`, but this would be fine in a script.

I would suggest leaving the `.draw()` functionality as is, and instead adding an `_ipython_display_` method to the `QuantumCircuit` class. Then if someone wants to view their circuit they can type `display(qc)`, or just `qc` if it's the last line of a cell. Slowly over time we can encourage the behaviour of using `display(<object>)` to show something in a notebook, and `.draw()` when we want to do something with that drawing.

**Comment by 1ucian0** [comment]
> I think this would cause confusing behaviour. For example, in Jupyter Notebooks we wouldn't be able to do `qc.draw('mpl').savefig()`, but this would be fine in a script.

What about `qc.draw(..., display=True)`?

**Comment by frankharkins** [comment]
> What about `qc.draw(..., display=True)`?

This is better, but notebook users typing `qc.draw(display=True)` at the end of a cell will see a drawing followed by `'None'`. I think it's still easier to keep existing functionality and encourage people to type `display(qc.draw())` for these cases.

File: qiskit/tools/jupyter/__init__.py [hunk 1]
```diff
@@ -106,7 +98,6 @@
 from .progressbar import HTMLProgressBar
 from .version_table import VersionTable
 from .copyright import Copyright
-from .monospace import MonospacedOutput
```
Comment (mtreinish) [inline qiskit/tools/jupyter/__init__.py:109]: This should be deprecated before being removed. We also just added this last release it's a pretty bad user experience to say "hey, here's a new thing you can use to adjust the text drawer output" Then 2-3 months later "that new thing is going away"

File: qiskit/circuit/quantumcircuit.py [hunk 2]
```diff
@@ -240,7 +246,8 @@ def data(self, data_input):
             self.append(inst, qargs, cargs)
 
     def __str__(self):
-        return str(self.draw(output='text'))
+        from qiskit.visualization import circuit_drawer
+        return str(circuit_drawer(self, output='text'))
```
Comment (mtreinish) [inline qiskit/circuit/quantumcircuit.py:250]: This change is pointing to why this is not a good idea. We don't want to change the return type based on the environment you run in.

File: qiskit/circuit/quantumcircuit.py [hunk 3]
```diff
@@ -1387,6 +1394,11 @@ def draw(self, output=None, scale=None, filename=None, style=None,
                               ax=ax,
                               initial_state=initial_state,
                               cregbundle=cregbundle)
+        if HAS_DISPLAY:
+            display(draw)
+            return None
+        else:
+            return draw
```
Comment (mtreinish) [inline qiskit/circuit/quantumcircuit.py:1401]: If you were making this change why only on the draw method and not circuit_drawer itself?

**Review by mtreinish — CHANGES_REQUESTED** [review]
I agree with @frankharkins here, we don't want to change the return type that will break user expectations for using draw(). I think having this be something we document for users to manually call in the jupyter context makes more sense.

**Comment by 1ucian0** [comment]
> I agree with @frankharkins here, we don't want to change the return type that will break user expectations for using draw().

Yeap... I agree. What about a `qc.display(...)`

**Comment by 1ucian0** [comment]
The right way to solve this is by calling `display` (by default in the jupyter notebook scope). 
```python
for circuit in circuits:
     display(circuit.draw(...))
```

**Pull request closed without merging**